upup/models/cloudup/resources/addons/node-termination-handler.aws/k8s-1.12.yaml.template

```diff
@@ -0,0 +1,176 @@
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: aws-node-termination-handler
+      app.kubernetes.io/instance: aws-node-termination-handler
+      kubernetes.io/os: linux
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: aws-node-termination-handler
+        app.kubernetes.io/instance: aws-node-termination-handler
+        k8s-app: aws-node-termination-handler
+        kubernetes.io/os: linux
+    spec:
+      volumes:
+        - name: "uptime"
+          hostPath:
+            path: "/proc/uptime"
+      priorityClassName: "system-node-critical"
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                - key: "kubernetes.io/os"
+                  operator: In
+                  values:
+                    - linux
+                - key: "kubernetes.io/arch"
+                  operator: In
+                  values:
+                    - amd64
+                    - arm64
+                    - arm
+      serviceAccountName: aws-node-termination-handler
+      hostNetwork: true
+      dnsPolicy: "ClusterFirstWithHostNet"
+      containers:
+        - name: aws-node-termination-handler
+          image: amazon/aws-node-termination-handler:v1.7.0
+          imagePullPolicy: IfNotPresent
+          securityContext:
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            runAsUser: 1000
+            runAsGroup: 1000
+            allowPrivilegeEscalation: false
+          volumeMounts:
+            - name: "uptime"
+              mountPath: "/proc/uptime"
+              readOnly: true
+          env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: SPOT_POD_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          - name: DELETE_LOCAL_DATA
+            value: "true"
+          - name: IGNORE_DAEMON_SETS
+            value: "true"
+          - name: POD_TERMINATION_GRACE_PERIOD
+            value: "-1"
+          - name: ENABLE_SPOT_INTERRUPTION_DRAINING
+            value: "{{ .EnableSpotInterruptionDraining }}"
+          - name: ENABLE_SCHEDULED_EVENT_DRAINING
+            value: "{{ .EnableScheduledEventDraining }}"
+          - name: JSON_LOGGING
+            value: "true"
+          - name: ENABLE_PROMETHEUS_SERVER
+            value: "{{ .EnablePrometheusMetrics }}"
+          resources:
+            limits:
+              memory: 128Mi
+            requests:
+              cpu: 50m
+              memory: 64Mi
+      nodeSelector:
```

Comment:
I think it would make sense to at least have the option of adding a `nodeSelector` that restricts it to "node-role.kubernetes.io/spot-worker": "true"

It might make sense to make that the default when scheduled event draining is disabled.

Reply:
Not against this one either. But perhaps as a follow-up?